### PR TITLE
Offload bad chunks

### PR DESCRIPTION
**Pull request by madmaxoft**
Chunks that fail to load are stored in separate files in a `badchunks` folder and regenerated.

### CODE REVIEW

**Comment by SphinxC0re** [comment]
This might collide with #2371

**Comment by madmaxoft** [comment]
It will most likely create one or two merge conflicts, but those should be easy to fix.

File: src/OSSupport/File.cpp [hunk 1]
```diff
@@ -172,7 +193,7 @@ int cFile::Write(const void * iBuffer, size_t iNumBytes)
 		return -1;
 	}
 
-	int res = (int)fwrite(iBuffer, 1, (size_t)iNumBytes, m_File);  // fwrite() returns the portion of Count parameter actually written, so we need to send iNumBytes as Count
+	int res = static_cast<int>(fwrite(a_Buffer, 1, a_NumBytes, m_File));  // fwrite() returns the portion of Count parameter actually written, so we need to send iNumBytes as Count
```
Comment (NiLSPACE) [inline src/OSSupport/File.cpp]: Comment on this line is wrong now. `iNumBytes` => `a_NumBytes`

Comment (madmaxoft) [inline src/OSSupport/File.cpp]: Good catch

**Comment by madmaxoft** [comment]
Done, rebased, I think this is now ready for merging.

**Pull request merged**